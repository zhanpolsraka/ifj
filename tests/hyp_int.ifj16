class Main{
	static int a = 3;
	static int b = 4;
	static void run(){
		int hypotenuse = hyp();
	}
	static void hyp(){
		int aa = Main.a*Main.a;
		int bb = Main.b*Main.b;
		int c = aa+bb;
		return c;
	}
}
