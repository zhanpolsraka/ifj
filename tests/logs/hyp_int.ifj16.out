---------------FILENAME: tests/hyp_int.ifj16---------------


Instruction stack item [24] -> class with name "Main"
Instruction stack item [23] -> instruction with operation 37 -> INSTR_ASS_ARG
Instruction stack item [22] -> instruction with operation 28 -> ASSIGNMENT
Instruction stack item [21] -> instruction with operation 37 -> INSTR_ASS_ARG
Instruction stack item [20] -> instruction with operation 28 -> ASSIGNMENT
Instruction stack item [19] -> function with name "run"
Instruction stack item [18] -> instruction with operation 35 -> INSTR_CALL_FUNC
Instruction stack item [17] -> instruction with operation 28 -> ASSIGNMENT
Instruction stack item [16] -> end function
Instruction stack item [15] -> function with name "hyp"
Instruction stack item [14] -> instruction with operation 37 -> INSTR_ASS_ARG
Instruction stack item [13] -> instruction with operation 37 -> INSTR_ASS_ARG
Instruction stack item [12] -> instruction with operation 3 -> MUL
Instruction stack item [11] -> instruction with operation 28 -> ASSIGNMENT
Instruction stack item [10] -> instruction with operation 37 -> INSTR_ASS_ARG
Instruction stack item [9] -> instruction with operation 37 -> INSTR_ASS_ARG
Instruction stack item [8] -> instruction with operation 3 -> MUL
Instruction stack item [7] -> instruction with operation 28 -> ASSIGNMENT
Instruction stack item [6] -> instruction with operation 37 -> INSTR_ASS_ARG
Instruction stack item [5] -> instruction with operation 37 -> INSTR_ASS_ARG
Instruction stack item [4] -> instruction with operation 1 -> PLUS
Instruction stack item [3] -> instruction with operation 28 -> ASSIGNMENT
Instruction stack item [2] -> instruction with operation 37 -> INSTR_ASS_ARG
Instruction stack item [1] -> end function
Instruction stack item [0] -> end class


---------------MAKE RELATIONS--------------
                     *
                     *
                     *
------------------EXECUTE------------------

------------------OUTPUT-------------------

STEP 1 - non-func instructions
Found Main
STEP 1 - non-func instructions
INSERT 3
---------------------------------------------------
N0: type is 22, values is 3
---------------------------------------------------
STEP 1 - non-func instructions
ASSIGN 3 to 0x233b400
---------------------------------------------------
N0: type is 22, values is 3
---------------------------------------------------
---------------------------------------------------
---------------------------------------------------
STEP 1 - non-func instructions
INSERT 4
---------------------------------------------------
N0: type is 22, values is 4
---------------------------------------------------
STEP 1 - non-func instructions
ASSIGN 4 to 0x233b4c0
---------------------------------------------------
N0: type is 22, values is 4
---------------------------------------------------
---------------------------------------------------
---------------------------------------------------
STEP 1 - non-func instructions
Got in function - ignore instructions until I get out
Found run()
STEP 1 - non-func instructions
STEP 1 - non-func instructions
STEP 1 - non-func instructions
Got out of function
Got out of run()
STEP 1 - non-func instructions
Got in function - ignore instructions until I get out
STEP 1 - non-func instructions
STEP 1 - non-func instructions
STEP 1 - non-func instructions
STEP 1 - non-func instructions
STEP 1 - non-func instructions
STEP 1 - non-func instructions
STEP 1 - non-func instructions
STEP 1 - non-func instructions
STEP 1 - non-func instructions
STEP 1 - non-func instructions
STEP 1 - non-func instructions
STEP 1 - non-func instructions
STEP 1 - non-func instructions
STEP 1 - non-func instructions
Got out of function
STEP 1 - non-func instructions
Got out of Main
STEP 2 - validating existance of run() in Main
STEP 3 - executing run() instructions
INSTR_CALL_FUNC
INSERT 3
---------------------------------------------------
N0: type is 22, values is 3
---------------------------------------------------
INSERT 3
---------------------------------------------------
N1: type is 22, values is 3
N0: type is 22, values is 3
---------------------------------------------------
UNKNOWN INSTRUCTION
---------------------------------------------------
N0: type is 22, values is 9
---------------------------------------------------
ASSIGN 9 to 0x233ba70
---------------------------------------------------
N0: type is 22, values is 9
---------------------------------------------------
---------------------------------------------------
---------------------------------------------------
INSERT 4
---------------------------------------------------
N0: type is 22, values is 4
---------------------------------------------------
INSERT 4
---------------------------------------------------
N1: type is 22, values is 4
N0: type is 22, values is 4
---------------------------------------------------
UNKNOWN INSTRUCTION
---------------------------------------------------
N0: type is 22, values is 16
---------------------------------------------------
ASSIGN 16 to 0x233b260
---------------------------------------------------
N0: type is 22, values is 16
---------------------------------------------------
---------------------------------------------------
---------------------------------------------------
INSERT 9
---------------------------------------------------
N0: type is 22, values is 9
---------------------------------------------------
INSERT 16
---------------------------------------------------
N1: type is 22, values is 16
N0: type is 22, values is 9
---------------------------------------------------
PLUS
---------------------------------------------------
N0: type is 22, values is 25
---------------------------------------------------
ASSIGN 25 to 0x233c3b0
---------------------------------------------------
N0: type is 22, values is 25
---------------------------------------------------
---------------------------------------------------
---------------------------------------------------
INSERT 25
---------------------------------------------------
N0: type is 22, values is 25
---------------------------------------------------
---------------------------------------------------
N0: type is 22, values is 25
---------------------------------------------------
ASSIGN 25 to 0x233b860
---------------------------------------------------
N0: type is 22, values is 25
---------------------------------------------------
---------------------------------------------------
---------------------------------------------------

-----------------END OUTPUT----------------


----------------ALL RIGHT!!!---------------

----------------SYMBOL TREE----------------

Class Main
	Variable -> a, type INTEGER, value [3]
	Variable -> b, type INTEGER, value [4]
	Function -> run
Lokal variables: 
	Variable -> hypotenuse, type INTEGER, value [25]
	Function -> hyp
Lokal variables: 
	Variable -> aa, type INTEGER, value [9]
	Variable -> bb, type INTEGER, value [16]
	Variable -> c, type INTEGER, value [25]


--------------INSTRUCTION LIST-------------

Instruction stack item [24] -> class with name "Main"
Instruction stack item [23] -> instruction with operation 37 -> INSTR_ASS_ARG
Instruction stack item [22] -> instruction with operation 28 -> ASSIGNMENT
Instruction stack item [21] -> instruction with operation 37 -> INSTR_ASS_ARG
Instruction stack item [20] -> instruction with operation 28 -> ASSIGNMENT
Instruction stack item [19] -> function with name "run"
Instruction stack item [18] -> instruction with operation 35 -> INSTR_CALL_FUNC
Instruction stack item [17] -> instruction with operation 28 -> ASSIGNMENT
Instruction stack item [16] -> end function
Instruction stack item [15] -> function with name "hyp"
Instruction stack item [14] -> instruction with operation 37 -> INSTR_ASS_ARG
Instruction stack item [13] -> instruction with operation 37 -> INSTR_ASS_ARG
Instruction stack item [12] -> instruction with operation 3 -> MUL
Instruction stack item [11] -> instruction with operation 28 -> ASSIGNMENT
Instruction stack item [10] -> instruction with operation 37 -> INSTR_ASS_ARG
Instruction stack item [9] -> instruction with operation 37 -> INSTR_ASS_ARG
Instruction stack item [8] -> instruction with operation 3 -> MUL
Instruction stack item [7] -> instruction with operation 28 -> ASSIGNMENT
Instruction stack item [6] -> instruction with operation 37 -> INSTR_ASS_ARG
Instruction stack item [5] -> instruction with operation 37 -> INSTR_ASS_ARG
Instruction stack item [4] -> instruction with operation 1 -> PLUS
Instruction stack item [3] -> instruction with operation 28 -> ASSIGNMENT
Instruction stack item [2] -> instruction with operation 37 -> INSTR_ASS_ARG
Instruction stack item [1] -> end function
Instruction stack item [0] -> end class



